domain-ee/ee-ep-merge-app/src/python_src/service/hoppy_service.py

```diff
@@ -43,17 +43,22 @@ def create_client(self, name):
     def get_client(self, name):
         return self.clients.get(name)
 
-    async def start_hoppy_clients(self, loop):
+    async def start_hoppy_clients(self):
+        loop = asyncio.get_event_loop()
         for client in self.clients.values():
             await client.start(loop)
-            publisher_connection = client.async_publisher._connection
-            consumer_connection = client.async_consumer._connection
-            while not publisher_connection.is_open or not consumer_connection.is_open:
+            while not client.is_ready():
                 await asyncio.sleep(0)
 
     async def stop_hoppy_clients(self):
         for client in self.clients.values():
             await client.stop()
 
+    def is_ready(self) -> bool:
+        ready = True
+        for client in self.clients.values():
+            ready = ready and client.is_ready()
+        return ready
```

Comment:
This could be simplified a bit with `all`:
```suggestion
        return all(client.is_ready() for client in self.clients.values())
```